src/ui/public/accessibility/__tests__/kbn_accessible_click.js

```diff
@@ -0,0 +1,120 @@
+import angular from 'angular';
+import sinon from 'auto-release-sinon';
+import expect from 'expect.js';
+import ngMock from 'ng_mock';
+import '../kbn_accessible_click';
+import {
+  ENTER_KEY,
+  SPACE_KEY,
+} from '../accessible_click_keys';
+
+describe('kbnAccessibleClick directive', () => {
+  let $compile;
+  let $rootScope;
+
+  beforeEach(ngMock.module('kibana'));
+
+  beforeEach(ngMock.inject(function (_$compile_, _$rootScope_) {
+    $compile = _$compile_;
+    $rootScope = _$rootScope_;
+  }));
+
+  describe('throws an error', () => {
+    it('when the element is a button', () => {
+      const html = `<button kbn-accessible-click></button>`;
+      expect(() => {
+        $compile(html)($rootScope);
+      }).to.throwError(`kbnAccessibleClick doesn't need to be used on a button.`);
+    });
+
+    it('when the element is a link with an href', () => {
+      const html = `<a href="#"" kbn-accessible-click></a>`;
+      expect(() => {
+        $compile(html)($rootScope);
+      }).to.throwError(`kbnAccessibleClick doesn't need to be used on a link if it has a href attribute.`);
+    });
+
+    it(`when the element doesn't have an ng-click`, () => {
+      const html = `<div kbn-accessible-click></div>`;
+      expect(() => {
+        $compile(html)($rootScope);
+      }).to.throwError(`kbnAccessibleClick requires ng-click to be defined on its element.`);
+    });
+  });
+
+  describe(`doesn't throw an error`, () => {
+    it('when the element is a link without an href', () => {
+      const html = `<a ng-click="noop" kbn-accessible-click></a>`;
+      expect(() => {
+        $compile(html)($rootScope);
+      }).not.to.throwError(`kbnAccessibleClick doesn't need to be used on a link if it has a href attribute.`);
```

Comment:
If you switch this to a regex it will blow up. It seems that it's not possible to test for **not** throwing a specific exception, so you should just pass 0 args here.